src/controllers/controller.bar.js

```diff
@@ -437,9 +437,12 @@ export default class BarController extends DatasetController {
       .filter(meta => meta.controller.options.grouped);
     const stacked = iScale.options.stacked;
     const stacks = [];
+    const currentParsed = this._cachedMeta.controller.getParsed(dataIndex);
+    const xStackIndex = currentParsed && currentParsed[iScale.axis];
```

Comment:
I think that `iScaleValue` might be a better name since this is not really the stack index but the value of the index scale

Reply:
Hi LeeLenaleee, x may not a good name. From the parsed data perspective, it's iScale value. But from the chart perspective, it's represent in which section. The 'RED' or the 'Blue' in the original issue example.
![image](https://github.com/user-attachments/assets/9b6d7b49-29c1-4c25-9e8f-434b780abd26)

Reply:
> I think that `iScaleValue` might be a better name since this is not really the stack index but the value of the index scale

Hi LeeLenaleee, x may not a good name. From the parsed data perspective, it's iScale value. But from the chart perspective, it's represent in which section. The 'RED' or the 'Blue' in the original issue example.
![image](https://github.com/user-attachments/assets/9b6d7b49-29c1-4c25-9e8f-434b780abd26)

Reply:
I would keep the code context here in mind more then how it shows on the chart itself. We read the property of the index scale which can also be the y axis. And internally it is a number and not the string you see at the chart. This conversion happens later.